src/core/ext/filters/client_channel/resolver/google_c2p/google_c2p_resolver.cc

```diff
@@ -308,11 +310,16 @@ void GoogleCloud2ProdResolver::StartXdsResolver() {
         {"TRAFFICDIRECTOR_DIRECTPATH_C2P_IPV6_CAPABLE", true},
     };
   }
+  std::string server_uri = "directpath-trafficdirector.googleapis.com";
+  if (g_traffic_director_uri_override != nullptr &&
```

Comment:
Since we have to check this anyway on every call to this function, why not just call `gpr_getenv()` here directly, so there's no need for the global variable at all?

The code here could just be:

```
UniquePtr<char> override_server(
    gpr_getenv("GRPC_TEST_ONLY_GOOGLE_C2P_RESOLVER_TRAFFIC_DIRECTOR_URI"));
const char* server_uri =
    override_server == nullptr
        ? "directpath-trafficdirector.googleapis.com"
        : override_server.get();
```

Reply:
done (tweaked the conditional though to check for zero-length string)